tests/unit/wait.tcl

```diff
@@ -179,33 +179,47 @@ tags {"wait aof network external:skip"} {
                 waitForBgrewriteaof $replica ;# Make sure there is no AOFRW
 
                 $master incr foo
-                assert_equal [$master waitaof 0 1 0] {1 1}
+                assert_equal [$master waitaof 0 1 5000] {1 1}
             }
 
             test {WAITAOF replica copy everysec with AOFRW} {
                 $replica config set appendfsync everysec
 
-                # Need an AOFRW that can be done in a second.
+                # When we trigger an AOFRW, a fsync is triggered when closing the old INCR file,
+                # so with the everysec, we will skip that second of fsync, and in the next second
+                # after that, we will eventually do the fsync.
                 $replica bgrewriteaof
 
                 $master incr foo
-                assert_equal [$master waitaof 0 1 0] {1 1}
+                assert_equal [$master waitaof 0 1 10000] {1 1}
+
+                waitForBgrewriteaof $replica
             }
 
-            test {WAITAOF replica copy everysec->always with AOFRW} {
-                $replica config set appendfsync everysec
+            test {WAITAOF replica copy everysec with slow AOFRW} {
+                $master set foo2 bar
+                $master set foo3 bar
+                wait_for_ofs_sync $master $replica
 
-                waitForBgrewriteaof $replica
+                $replica config set appendfsync everysec
+                $replica config set rdb-key-save-delay 1000000 ;# 1 sec
 
-                # Need an AOFRW that can be done in a second.
+                # There are three keys in total, so AOFRW takes three seconds to complete.
```

Comment:
maybe making it take 1 second is enough?
maybe combined with validating that WAITAOF returned before the rewrite completed?